### PR TITLE
Validate input range of `du` flags

### LINKED ISSUE #7953: du crashes nushell with invalid flag value
### Describe the bug

```sh
du -d -1
thread 'main' panicked at 'error converting i64 to u64: TryFromIntError(())', /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/nu-command-0.75.0/src/platform/du.rs:95:48
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

### How to reproduce

execute `du -d -1`

### Expected behavior

Not crash `nu` and just print an error

### Screenshots

_No response_

### Configuration

| key                | value                                                      |
| ------------------ | ---------------------------------------------------------- |
| version            | 0.75.0                                                     |
| branch             |                                                            |
| commit_hash        |                                                            |
| build_os           | linux-x86_64                                               |
| build_target       | x86_64-unknown-linux-gnu                                   |
| rust_version       | rustc 1.69.0-nightly (11d96b593 2023-02-01)                |
| rust_channel       | nightly-x86_64-unknown-linux-gnu                           |
| cargo_version      | cargo 1.69.0-nightly (e84a7928d 2023-01-31)                |
| pkg_version        | 0.75.0                                                     |
| build_time         | 2023-02-03 07:34:54 +01:00                                 |
| build_rust_channel | release                                                    |
| features           | database, dataframe, default, trash, which, zip            |
| installed_plugins  | from plist, gstat, query, query json, query web, query xml |

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Mehrbod2002**
# Description
Fix #7953
Fix flags on `du` which should be positive and added some tests. 

# User-Facing Changes

_(List of all changes that impact the user experience here. This helps us keep track of breaking changes.)_

# Tests + Formatting

Don't forget to add tests that cover your changes.

Make sure you've run and fixed any issues with these commands:

- `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- `cargo clippy --workspace -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- `cargo test --workspace` to check that all tests pass

# After Submitting

If your PR had any user-facing changes, update [the documentation](https://github.com/nushell/nushell.github.io) after the PR is merged, if necessary. This will help us keep the docs up to date.

### CODE REVIEW

**Comment by fdncred** [comment]
Is this better than #7960?

**Comment by Mehrbod2002** [comment]
> Is this better than #7960?

Not going to judge it . but believe as it's going to convert it u64 it means need positive value and have to be sure with it , just this .

**Comment by fdncred** [comment]
Sorry, I wasn't trying to compare the two PRs. Was just trying to decide which fix we should accept since they both seem to fix the same problem but in different ways.

**Comment by Mehrbod2002** [comment]
> Sorry, I wasn't trying to compare the two PRs. Was just trying to decide which fix we should accept since they both seem to fix the same problem but in different ways.

I got what is your intention. No need for  apologies. 🙏🤜🤛

File: crates/nu-protocol/src/ast/call.rs [hunk 1]
```diff
@@ -130,6 +130,18 @@ impl Call {
...
+        for name in self.named_iter() {
+            if flag_name == name.0.item {
+                if let Some(val) = &name.2 {
+                    return Some(val.span);
+                }
+            }
+        }
+
+        None
+    }
```
Comment (sholderbach) [inline crates/nu-protocol/src/ast/call.rs:143]: While I see the usefulnesss of this little helper and there is nothing inherently wrong with it, it feels a bit odd to add it for just two uses. `nu_engine::CallExt::get_flag<Spanned<T: FromValue>>` is probably what is used in most places.

File: crates/nu-command/tests/commands/platform/du.rs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+#[test]
+fn test_du_flag_max_depth() {
+    let actual = nu!("du -d -2");
+    assert!(actual
+        .err
+        .contains("Negative value passed when positive one is required"));
+
+    let actual = nu!("du -d 2");
+    assert!(actual.err.is_empty());
```
Comment (sholderbach) [inline crates/nu-command/tests/commands/platform/du.rs:22]: Thanks for the tests. Do we need to constrain what files `du` will try to look at? It shouldn't traverse half the file system for a small unit test of flags.

**Review by sholderbach — COMMENTED** [review]
Thanks for the defensive error checking!

File: crates/nu-command/tests/commands/platform/du.rs (hunk 2 above)
Comment (Mehrbod2002) [reply crates/nu-command/tests/commands/platform/du.rs:22]: How can we do that ? 
Shall we remove successfull result tests ?

File: crates/nu-protocol/src/ast/call.rs (hunk 1 above)
Comment (Mehrbod2002) [reply crates/nu-protocol/src/ast/call.rs:143]: I didn't know that 
Will resolve on next commit 
Thanks

File: crates/nu-command/tests/commands/platform/du.rs (hunk 2 above)
Comment (sholderbach) [reply crates/nu-command/tests/commands/platform/du.rs:22]: Either that or pointing the `cwd` to a directory with not much in it.

Comment (Mehrbod2002) [reply crates/nu-command/tests/commands/platform/du.rs:22]: Got it 
Will resolve

**Pull request merged**